### PR TITLE
Extract e2e stories

### LINKED ISSUE #1215: enable extraction of e2e stories from trackerstore
For getting out conversation logs and creating tests, it would be useful if we could extract conversations from the trackerstore as end-to-end stories

### PR DESCRIPTION
**Pull request by amn41**
**Proposed changes**:
- fixes #1215 

**Status (please check what you already did)**:
- [ ] made PR ready for code review
- [x] added some tests for the functionality
- [ ] updated the documentation
- [ ] updated the changelog

### CODE REVIEW

File: rasa_core/events/__init__.py [hunk 1]
```diff
@@ -230,17 +230,21 @@ def _from_story_string(cls, parameters):
...
                 ent_string = ""
 
-            return "{intent}{entities}".format(
-                    intent=self.intent.get("name", ""),
-                    entities=ent_string)
+            parse_string = "{intent}{entities}".format(
+                           intent=self.intent.get("name", ""),
+                           entities=ent_string)
+            if e2e:
+                return "{}:{}".format(parse_string, self.text)
```
Comment (ricwo) [inline rasa_core/events/__init__.py]: The end-to-end format we've been using so far uses md-style messages that include entities:
`* inform: on the [south](location) part of town`

Can we use `rasa_core.evaluate._md_format_message()` and maybe move it here?

File: rasa_core/server.py [hunk 2]
```diff
@@ -330,6 +330,35 @@ def retrieve_tracker(sender_id):
...
+        # retrieve tracker and set to requested state
+        tracker = agent.tracker_store.get_or_create_tracker(sender_id)
+        if not tracker:
+            return error(503,
+                         "NoDomain",
+                         "Could not retrieve tracker. Most likely "
+                         "because there is no domain set on the agent.")
+
+        if until_time is not None:
+            tracker = tracker.travel_back_in_time(float(until_time))
```
Comment (ricwo) [inline rasa_core/server.py]: might be worth `try`ing to `float`?

**Comment by amn41** [comment]
cool thanks @ricwo please have another look

File: docs/_static/spec/server.yml [hunk 3]
```diff
@@ -893,6 +927,18 @@ components:
...
 
+    200Story:
+      description: Success
+      content:
+        text/markdown:
+          example: >-
+            ## story_00055028
+            * greet: hello
+             - utter_ask_howcanhelp
+            * inform{"cuisine": "indian", "people": "two", "price": "moderate"}: I'm looking for a moderately priced Indian restaurant for two people
```
Comment (ricwo) [inline docs/_static/spec/server.yml]: isn't this a different format? `md_format_message()` should generate something like `*inform: I'm looking for a [moderately priced](price:moderate)...`

**Pull request merged**